internal/httpclient/policies.go

```diff
@@ -0,0 +1,204 @@
+package httpclient
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/julienstroheker/AzHexGate/internal/logging"
+)
+
+// RequestIDPolicy adds a unique request ID to each request
+type RequestIDPolicy struct {
+	headerName string
+}
+
+// NewRequestIDPolicy creates a new RequestIDPolicy
+func NewRequestIDPolicy(headerName string) *RequestIDPolicy {
+	if headerName == "" {
+		headerName = "X-Client-Request-Id"
+	}
+	return &RequestIDPolicy{headerName: headerName}
+}
+
+// Do implements Policy interface
+func (p *RequestIDPolicy) Do(
+	req *http.Request,
+	next func(*http.Request) (*http.Response, error),
+) (*http.Response, error) {
+	// Generate a new UUID for this request
+	requestID := uuid.New().String()
+	req.Header.Set(p.headerName, requestID)
+	return next(req)
+}
+
+// LoggingPolicy logs requests and responses in debug mode
+type LoggingPolicy struct {
+	logger *logging.Logger
+}
+
+// NewLoggingPolicy creates a new LoggingPolicy
+func NewLoggingPolicy(logger *logging.Logger) *LoggingPolicy {
+	return &LoggingPolicy{logger: logger}
+}
+
+// Do implements Policy interface
+func (p *LoggingPolicy) Do(
+	req *http.Request,
+	next func(*http.Request) (*http.Response, error),
+) (*http.Response, error) {
+	if p.logger == nil {
+		return next(req)
+	}
+
+	// Log request
+	p.logger.Debug("HTTP Request",
+		logging.String("method", req.Method),
+		logging.String("url", req.URL.String()),
+		logging.String("request_id", req.Header.Get("X-Client-Request-Id")))
```

Comment:
We should not assume this header to be present to keep it generic, we should log all headers if there is some.